src/dodal/devices/apple2_undulator.py

```diff
@@ -391,19 +406,19 @@ def __init__(
         """
         self.update_lookuptable()
 
-    def check_pol(self, pol: str) -> bool:
-        if pol in self._available_pol:
-            return True
-        else:
-            raise ValueError(
-                f"Polarisation {pol} is not available:"
-                + f"/n Polarisations available:  {self._available_pol}"
-            )
-
-    def set_pol(self, pol: str) -> None:
+    # Hardward backed polarisation function,
+    async def read_pol_setpoint(self):
+        pol = await self.polarisation_setpoint.get_value()
+        # LH3 is not hardware readable as it is the same as lh but it is needed for energy.
+        if pol.value != "lh3":
```

Comment:
Could: If you have the enum now you can use it:
```suggestion
        if pol != Pol.LH3:
```